### PR TITLE
[EXAMPLE] Adds test framework via `vitest`

### PR DESCRIPTION
**Pull request by kieraneglin**
This is for a conversation I'm having with Zakary - can be ignored for now (:

### CODE REVIEW

**Review by mmgoodnow — COMMENTED** [review]
This will be super valuable for our release parsing code especially, which is very hard to test for regressions manually.

File: tests/utils.test.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  })
+
+  describe('when testing for video files by extension', () => {
+    it('returns MediaType.MOVIE if the title matches MOVIE_REGEX', () => {
+      const file = { name: 'media.mp4', length: 1, path: '/tmp/media.mp4' } as File
+      expect(getMediaType({ title: 'My.Movie.2021', files: [file] } as Searchee)).toBe(MediaType.MOVIE)
+    })
+
+    it('returns MediaType.ANIME if the title matches ANIME_REGEX', () => {
+      const file = { name: 'media.mp4', length: 1, path: '/tmp/media.mp4' } as File
```
Comment (mmgoodnow) [inline tests/utils.test.ts:34]: why all the `as`es?

Comment (mmgoodnow) [reply tests/utils.test.ts]: looks good!

**Comment by mmgoodnow** [comment]
Anything else you plan to do before making this ready for review, or?

**Comment by zakkarry** [comment]
I mean I was going to get some test cases for the regex's implemented....I suppose as a framework we can take it out of draft if you want to incrementally add tests and just get this one done for regex for now on the basic release regexs.

IIRC @ShanaryS has a bunch of test cases, might have only been anime, and I can always get a list of my directories and torrent file names and shit for TV and Movies.

I guess the real question is, at what level do you want this before its merged?

**Comment by mmgoodnow** [comment]
Well any tests are beneficial, so I'd be happy to merge it as-is.

**Comment by mmgoodnow** [comment]
should be able to just add `npm test` to this line https://github.com/cross-seed/cross-seed/blob/master/.github/workflows/ci.yml#L27

**Review by mmgoodnow — APPROVED** [review]

**Comment by mmgoodnow** [comment]
Thanks for the contribution!

**Pull request merged**